modules/calib3d/src/calibration.cpp

```diff
@@ -2840,11 +2837,15 @@ void cvGetOptimalNewCameraMatrix( const CvMat* cameraMatrix, const CvMat* distCo
         if( validPixROI )
         {
             icvGetRectangles( cameraMatrix, distCoeffs, 0, &matM, imgSize, inner, outer );
-            cv::Rect r = inner;
-            r &= cv::Rect(0, 0, newImgSize.width, newImgSize.height);
-            *validPixROI = cvRect(r);
         }
     }
+    if( validPixROI )
+    {
+        cv::Rect r(cvFloor(inner.x), cvFloor(inner.y), cvCeil(inner.x+inner.width)-cvFloor(inner.x)+1,
```

Comment:
I got the issue in `cv::Rect r = inner;` with rounding, just truncate. Why do you add +1?

Reply:
There was indeed an error. I rewrote that piece of code with a tolerance to get tests to pass.